tools/generate_events.py

```diff
@@ -90,15 +90,25 @@
     {%- endif %}
 }
 
+{%- if event.type == "cluster" %}
 func (e *{{eventName}}) GetClusterId() strfmt.UUID {
     return e.ClusterId
 }
-
-{%- if event.type != "cluster" %}
+{%- else %}
+func (e *{{eventName}}) GetClusterId() *strfmt.UUID {
+    {% if event.properties.cluster_id -%}     return &e.ClusterId
```

Comment:
it would be safer in this context to have a  nil-check on the e.ClusterId, since the user might provide nil as clusterID value and that will cause this call to fail.